target-app/Onboarding/Views/SignInViewController.swift

```diff
@@ -91,15 +91,44 @@ class SignInViewController: UIViewController {
     }
     
     init(viewModel: SignInViewModel) {
-      self.viewModel = viewModel
-      super.init(nibName: nil, bundle: nil)
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+        
+        setupBinders()
     }
     
     @available(*, unavailable)
     required init?(coder: NSCoder) {
       fatalError("init(coder:) has not been implemented")
     }
     
+    // MARK: - Binders
+    private var cancellables: Set<AnyCancellable> = []
+    
+    private func setupBinders() {
+        viewModel.statePublisher
+            .receive(on: RunLoop.main)
+            .sink { [weak self] state in
+                self?.renderState()
```

Comment:
you could pass the state as a parameter to this function to make it independent of the view model and make private the variable in the view model

Reply:
done!